File: src/main/java/org/distril/beengine/command/parser/TargetParser.java
package org.distril.beengine.command.parser;

import org.distril.beengine.command.CommandSender;

public class TargetParser extends Parser {

	public static final TargetParser INSTANCE = new TargetParser();

	@Override
	public String parse(CommandSender sender, String input) {
		if (input.equals("@s") && !sender.isConsole()) {
			return sender.getName();
		}

		return input;
	}
}

File: src/main/kotlin/org/distril/beengine/command/parser/TargetParser.kt
package org.distril.beengine.command.parser

import org.distril.beengine.command.CommandSender

object TargetParser : Parser() {

    override fun parse(sender: CommandSender, input: String) =
        if (input == "@s" && !sender.isConsole) sender.name else input
}
